### PR TITLE
Accelerated DAG performance test fixes

### LINKED ISSUE #45891: Release test microbenchmark_unstable failed
Release test **microbenchmark_unstable** failed. See https://buildkite.com/ray-project/release/builds/17343#01900afb-97ce-4b00-9837-05fa76e74b02 for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by jackhumphries**
## Why are these changes needed?

The accelerated DAG performance test code in `python/ray/_private/ray_experimental_perf.py` is out of sync with the accelerated DAG library. Thus, there were various runtime issues. This PR fixes those issues.

## Related issue number

Closes #45891

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/serialization.py [hunk 1]
```diff
@@ -493,10 +528,38 @@ def _python_serializer(o):
         msgpack_data = MessagePackSerializer.dumps(value, _python_serializer)
 
         if python_objects:
-            metadata = ray_constants.OBJECT_METADATA_TYPE_PYTHON
-            pickle5_serialized_object = self._serialize_to_pickle5(
-                metadata, python_objects
-            )
+            contains_adag = False
```
Comment (stephanie-wang) [inline python/ray/_private/serialization.py:531]: This seems like it was added by mistake?

**Review by stephanie-wang — COMMENTED** [review]
@jackhumphries are you still working on this PR?

File: python/ray/_private/serialization.py (hunk 1 above)
Comment (jackhumphries) [reply python/ray/_private/serialization.py]: Yep, thanks.

**Comment by jackhumphries** [comment]
@stephanie-wang This is ready for a review. Thanks Stephanie!

**Review by stephanie-wang — APPROVED** [review]

**Pull request merged**